hooks/utils/getInfiniteQuery.ts

```diff
@@ -0,0 +1,58 @@
+import { productAPI, userAPI } from 'apis';
+import { queryKeyType } from 'types/infiniteQueryKey';
+import { ProductsResponseType } from 'types/product';
+
+type getQueryApiByQueryKeyReturnType = ({
+  pageParam,
+}: {
+  pageParam?: number;
+}) => Promise<{
+  data: ProductsResponseType;
+  currentPage: number;
+  isLast: boolean;
+}>;
+
+// @TODO 중복 코드 제거 고민 필요
+const getProductsAPI = async ({ pageParam = 0 }) => {
+  const { data } = await productAPI.getProducts({ offset: pageParam });
+
+  return {
+    data,
+    currentPage: pageParam,
+    isLast: data.length ? false : true,
+  };
```

Comment:
아래처럼 함수를 만들어서 사용할 수 있을 것 같아요!
```js
const getReturnData = (data, pageParam) => {
  return {
    data,
    currentPage: pageParam,
    isLast: data.length ? false : true,
  }
}
```

```suggestion
  return getReturnData(data, pargeParam)
```

하지만,, 함수를 분리해도 결국 결합도가 높아지기 때문에 무조건 분리하는 것이 좋은 것은 아니라고 생각이 자꾸 들어서 조금 더 고민해보겠습니다ㅠ

Reply:
아니면 여기서는 `data`만 반환하고

`getQueryApiByQueryKey`에서 함수를 만들어줄 때 반환값을 만들어주는건 어떨까요?